snap/snapcraft.yaml

```diff
@@ -149,15 +149,17 @@ parts:
       rustup default 1.79
     override-build: |
       uv venv --relocatable $CRAFT_PART_INSTALL
-      uv sync --no-dev --no-editable --extra apt-jammy
+      uv sync --no-dev --no-editable --no-binary
+      . $CRAFT_PART_INSTALL/bin/activate
+      uv pip install -r requirements-jammy.txt
       rm $CRAFT_PART_INSTALL/lib64
 
       cp -v completion.bash $CRAFT_PART_INSTALL
 
       mkdir -p $CRAFT_PART_INSTALL/libexec/charmcraft
       sed -i 's|#!/bin/sh|#!/snap/charmcraft/current/bin/python3|' $CRAFT_PART_INSTALL/bin/craftctl
 
-      version="$($CRAFT_PART_INSTALL/bin/charmcraft --version | cut -d' ' -f2)"
+      version="$(uv pip show charmcraft|grep "Version:" | cut -d' ' -f2)"
```

Comment:
```suggestion
      version="$(uv pip show charmcraft | grep "Version:" | cut -d' ' -f2)"
```

Reply:
This is why we need a linter for snapcraft.yaml files.